webpack/assets/javascripts/react_app/components/ForemanModal/subcomponents/SubmitOrCancel/index.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { submitModal } from './SubmitOrCancelActions';
+import SubmitOrCancel from './SubmitOrCancel';
+
+const ConnectedSubmitOrCancel = ({ submitting, onCancel, submitProps, id }) => {
+  const dispatch = useDispatch();
+
+  const { submitBtnProps, cancelBtnProps, ...rest } = submitProps;
+
+  const boundOnSubmit = () =>
+    dispatch(
+      submitModal({
+        ...rest,
+        closeFn: onCancel,
+        id,
+      })
+    );
+
+  return (
+    <SubmitOrCancel
+      submitting={submitting}
+      onCancel={onCancel}
+      onSubmit={boundOnSubmit}
+      submitBtnProps={submitBtnProps}
+      cancelBtnProps={cancelBtnProps}
+    />
```

Comment:
I wonder if we should move the `SubmitOrCancel.js` code to this file, instead of passing props to both submit and cancel even if usually we have only one main component in `index.js`, e.g: 
```js
  <React.Fragment>
    <SubmitBtn
      onSubmit={onSubmit}
      submitting={submitting}
      {...submitBtnProps}
    />
    <CancelBtn onCancel={onCancel} disabled={submitting} {...cancelBtnProps} />
  </React.Fragment>
```

Reply:
SubmitOrCancel/index.js exports the 'connected' component (using redux hooks though, so 'hooked component'?), we would need to wrap it in `<Provider>` for tests.